src/connector/src/sink/clickhouse.rs

```diff
@@ -567,16 +606,30 @@ impl ClickHouseSinkWriter {
                             ClickHouseField::Int8(1),
                         ));
                     }
+                    if self.clickhouse_engine.get_delete_col().is_some() {
+                        clickhouse_filed_vec.push(ClickHouseFieldWithNull::WithoutSome(
+                            ClickHouseField::Int8(0),
+                        ))
+                    }
                 }
                 Op::Delete | Op::UpdateDelete => {
-                    if !self.clickhouse_engine.is_collapsing_engine() {
+                    if !self.clickhouse_engine.is_collapsing_engine()
+                        && !self.clickhouse_engine.is_delete_replacing_engine()
+                    {
                         return Err(SinkError::ClickHouse(
                             "Clickhouse engine don't support upsert".to_string(),
                         ));
                     }
-                    clickhouse_filed_vec.push(ClickHouseFieldWithNull::WithoutSome(
-                        ClickHouseField::Int8(-1),
-                    ))
+                    if self.clickhouse_engine.get_sign_name().is_some() {
+                        clickhouse_filed_vec.push(ClickHouseFieldWithNull::WithoutSome(
+                            ClickHouseField::Int8(-1),
```

Comment:
So do they have different values to represent insert and delete(-1 and 1 / 0 and 1), is this specified by ck?

Reply:
Correct, per their documentation 

> `is_deleted` — Name of a column used during a merge to determine whether the data in this row represents the state or is to be deleted; 1 is a "deleted" row, 0 is a "state" row.

Reply:
I feel like calling `.get_sign_name().is_some()` and `get_delete_column().is_some()` is performing some unnecessary memory copies on a per-row basis given both call a copy function (`.to_string()` and `.clone()`). 

I'm going to make a patch to remove this